sale_coupon_multiplier_free_product/models/sale_order.py

```diff
@@ -141,9 +142,10 @@ def _remove_invalid_reward_lines(self):
                 and x.coupon_program_id in multiple_of_programs_to_remove
             ).unlink()
         # We'll catch the context in the subsequent unlink() method
-        super(
+        res = super(
             SaleOrder, self.with_context(valid_multiple_of_lines=valid_lines)
         )._remove_invalid_reward_lines()
+        return res
```

Comment:
You don't need to make an assignment if you're going to return that value immediately after it
```suggestion
        return super(
            SaleOrder, self.with_context(valid_multiple_of_lines=valid_lines)
        )._remove_invalid_reward_lines()
```

Reply:
All changes done, thank you for all the suggestions.